src/Compilers/VisualBasic/Portable/Symbols/NamedTypeSymbol.vb

```diff
@@ -1150,7 +1150,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Dim baseAllRequiredMembers = If(BaseTypeNoUseSiteDiagnostics?.AllRequiredMembers, ImmutableSegmentedDictionary(Of String, Symbol).Empty)
             Dim typeHasDeclaredRequiredMembers = HasAnyDeclaredRequiredMembers
 
-            If (Not typeHasDeclaredRequiredMembers) AndAlso baseAllRequiredMembers.IsEmpty Then
+            ' Fast check: avoid realizing GetMembersUnordered() for substituted symbols when the original
+            ' definition doesn't contain any required members or any members with the same name as a required
+            ' member from the base type.
+            If Not ReferenceEquals(OriginalDefinition, Me) AndAlso
+                Not OriginalDefinition.GetMembersUnordered().Any(Function(member, capturedBaseAllRequiredMembers) member.IsRequired() OrElse capturedBaseAllRequiredMembers.ContainsKey(member.Name), baseAllRequiredMembers) Then
```

Comment:
>Not OriginalDefinition.GetMembersUnordered().Any(Function(member, capturedBaseAllRequiredMembers) member.IsRequired() OrElse capturedBaseAllRequiredMembers.ContainsKey(member.Name), baseAllRequiredMembers) Then [](http://example.com/codeflow?start=16&length=210)

The same comment about logic duplication as for C#.

Reply:
The requested alteration was straightforward for C# symbols. However, it doesn't look like the `SymbolAsMember` extension is available for VB symbols.

Reply:
It is called differently - ```AsMember```.

```
        <Extension>
        Friend Function AsMember(Of T As Symbol)(origMember As T, type As NamedTypeSymbol) As T
            Debug.Assert(origMember.IsDefinition)

            ' ContainingType is always a definition here which is a type definition so we can use "Is"
            If type Is origMember.ContainingType Then
                Return origMember
            End If

            Dim substituted = DirectCast(type, SubstitutedNamedType)
            Return DirectCast(substituted.GetMemberForDefinition(origMember), T)
        End Function
```

Reply:
➡️ Thank you, this is now updated.